### DISCUSSION TITLE
PaperWM Gnome 40 Discussions

### DISCUSSION THREAD

**Discussion by michaelhthomas** [PaperWM-community/PaperWM#1] [Announcements]
Hi there! 

Here's the place to hold discussions surrounding the development of fixes for PaperWM to support Gnome 40+ present in this fork, as well as the potential to contribute these fixes back to the main repo. Any discussion not specific to this fork should go in the [upstream issue](https://github.com/paperwm/PaperWM/issues/376).

**Comment by smichel17** [comment]
I'm worried that splitting the discussion will split the community & make it more difficult to merge this effort back upstream. (Especially since @hedning showed up a week ago to merge https://github.com/paperwm/PaperWM/pull/383).

I think it might be best to continue using issues in the main repo for this. I think `[Gnome 40]` in the title or something like that would be sufficient to keep track of issues for now. (If we get to the point where it's not enough, then it's probably time to consider this a hard fork, not a soft one.)

**Reply by michaelhthomas** [reply]
You make a good point. I'll leave this as a place for people to discuss things specific to this fork and getting it contributed back to the main repo, but we should probably keep issue / project related discussion in the main issue.

**Reply by smichel17** [reply]
Okay. I agree that this was useful — #4 would have been noise on the main repo. I don't think we need both issues and discussions, though. I'll disable issues for now (easy enough to enable them later if we need them; hard/annoying to disable once we start using them).

**Comment by paradajz** [comment]
There are a few patches on develop which aren't in next-release, one of them being my patch:

https://github.com/paperwm/PaperWM/commit/7331b44807f94d8a2d6adac6e75b0e32b2ad8caf
https://github.com/paperwm/PaperWM/commit/10215f57e8b34a044e10b7407cac8fac4b93bbbc
https://github.com/paperwm/PaperWM/commit/68e0ef7d9a9b7e2b272472518870d28be2f593fd

I am not sure if anything before my patch needs to be ported over to next-develop since develop and next-release aren't in sync and I can't figure out where they diverged.

**Reply by smichel17** [reply]
> develop and next-release aren't in sync and I can't figure out where they diverged.

I can take a swing at figuring that out (though maybe not for a few days, with the holidays and also a fairly long list of other tasks I want to get to soon). I don't know the gnome-shell code well enough to contribute effectively (and learning isn't my priority), but I do know js and git, so this is something I *can* tackle.
